diffrax/_integrate.py

```diff
@@ -137,7 +139,17 @@ def _check(term_cls, term, yi):
                 )
                 if not vf_type_compatible:
                     raise ValueError
-                control_type = jax.eval_shape(term.contr, 0.0, 0.0)
+
+                try:
+                    flag = issubclass(control_type_expected, AbstractBrownianIncrement)
+                except TypeError:
+                    flag = False
+
+                if flag:
+                    contr = functools.partial(term.contr, use_levy=True)
+                else:
+                    contr = term.contr
+                control_type = jax.eval_shape(contr, 0.0, 0.0)
```

Comment:
Ah, I see what your'e doing here. Hmm, this is kind of annoying. Perhaps we should add:
```python
class AbstractSolver(eqx.Module):
    term_compatible_contr_kwargs = dict()

class AbstractSRK(AbstractSolver):
    term_compatible_contr_kwargs = dict(use_levy=True)
```
and then use `jax.eval_shape(ft.partial(term.contr, **solver.term_compatible_contr_kwargs), 0.0, 0.0)` here.

Which is a terrible terrible hack (it would have ben better if we'd just had `contr` always return a single type, but backward compatibility makes that impossible now), but I think better than special-casing support here -- this would give us an extensible way to handle this problem again in the future.

Reply:
Oh, I see. Even if it is a hack, it's a much nicer one than my little patch. I'll fix it.